application/core/LSMessageSource.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/**
+ * LSMessageSource class file.
+ *
+ * @author Denis Chenu
+ * @link http://www.yiiframework.com/
+ * @copyright 2021 LimeSurvey
+ * License: GNU/GPL License v3 or later, see LICENSE.php
+ */
+
+class LSMessageSource extends CMessageSource
+{
+    public const CACHE_KEY_PREFIX = 'Yii.LSMessageSource.';
+    public const MO_FILE_EXT = '.mo';
+    public const PO_FILE_EXT = '.po';
+    /**
+     * @var string the ID of the database connection application component. Defaults to 'db'.
+     * @see CDbMessageSource::connectionID
+     */
+    public $connectionID = 'db';
+    /**
+     * @var string the name of the source message table. Defaults to 'SourceMessage'.
+     * The DB prefix is added automatically
+     * @see CDbMessageSource::sourceMessageTable
+     */
+    public $sourceMessageTable = 'sourcemessage';
+    /**
+     * @var string the name of the translated message table. Defaults to 'Message'.
+     * The DB prefix is added automatically
+     * @see CDbMessageSource::translatedMessageTable
+     */
+    public $translatedMessageTable = 'message';
+    /**
+     * @var integer the time in seconds that the messages can remain valid in cache.
+     * @see CDbMessageSource::cachingDuration
+     * @see CGettextMessageSource::cachingDuration
+     */
+    public $cachingDuration = 0;
+    /**
+     * @var string the ID of the cache application component that is used to cache the messages.
+     * Defaults to 'cache' which refers to the primary cache application component.
+     * @see CDbMessageSource::cacheID
+     * @see CGettextMessageSource::cacheID
+     */
+    public $cacheID = 'cache';
+    /**
+     * @var string the base path for all translated messages. Defaults to null, meaning
+     * @see CGettextMessageSource::basePath
+     */
+    public $basePath;
+    /**
+     * @var boolean whether to load messages from MO files. Defaults to true.
+     * @see CGettextMessageSource::useMoFile
+     */
+    public $useMoFile = true;
+    /**
+     * @var boolean whether to use Big Endian to read MO files.
+     * Defaults to false. This property is only used when {@link useMoFile} is true.
+     * see CGettextMessageSource::useBigEndian
+     */
+    public $useBigEndian = false;
+    /**
+     * @var string the message catalog name. This is the name of the message file (without extension)
+     * that stores the translated messages. Defaults to 'messages'.
+     * see CGettextMessageSource::catalog
+     */
+    public $catalog = 'messages';
+
+    /**
+     * Used as static for get DB
+     */
+    private $messagedb;
```

Comment:
Could this be moved to static variable in the function instead?

Reply:
See https://github.com/LimeSurvey/LimeSurvey/blob/fca5dc1256faf9c740eb5b0b7efc1daab428ea89/framework/i18n/CDbMessageSource.php#L94 

But yes only used inside this function

Reply:
PS : maybe just use current DB : not allowing to update ?

Reply:
removs : not allowing update by DB (keep it for a new plugin event ;) )